.circleci/config.yml

```diff
@@ -1105,19 +1114,20 @@ jobs:
   # Builds in C++20 mode and uses debug build in order to speed up.
   # Do *NOT* store any artifacts or workspace as we don't run tests on this build.
   b_ubu_cxx20:
-    <<: *base_ubuntu2404_xlarge
+    <<: *base_ubuntu2404_large
     environment:
-      <<: *base_ubuntu2404_xlarge_env
+      <<: *base_ubuntu2404_large_env
       CMAKE_BUILD_TYPE: Debug
-      CMAKE_OPTIONS: -DCMAKE_CXX_STANDARD=20
-      MAKEFLAGS: -j 10
+      # gold is more memory-efficient than the default, allows us to stay with the "large" resource class without OOM errors
+      CMAKE_OPTIONS: -DCMAKE_CXX_STANDARD=20 -DCMAKE_CXX_FLAGS=-fuse-ld=gold
```

Comment:
I think its better to not use `CMAKE_CXX_FLAGS` here, `CMAKE_EXE_LINKER_FLAGS` is better because those are linker flags. The problem is that not all compilers to define linker flags here. I know its our CI with our configured compilers and everything, but its cleaner and should produce the same results.